src/api.rs

```diff
@@ -179,8 +179,10 @@ pub enum WatchEvent {
     },
 }
 
-/// User identifier that will be passed from the REST API handlers to the
-/// adapters.
+/// User identifier that will be given to adapter's getters and setters if such
+/// identification is required for the specific request.
+/// For REST API requests, the user identifier is taken from the authentication
+/// header.
 #[derive(Debug, Clone, PartialEq)]
 pub enum User {
```

Comment:
Can you explain the case `Id(i32)`? It's not clear at all what this `i32` is.

Also, it might be clearer to have something along the lines of

```rust
pub struct User(pub i32);

/// ...

fn send_values(&self, TargetMap<SetterSelector, Value>, user: Option<User>) -> ResultMap<Id<Setter>, (), Error>;
```